### ISSUE TITLE
Dodging while entering stage animation misspositions said animation

### ISSUE THREAD

**Issue by ThatExplo** [studio-minus/madness-interactive-reloaded#220]
**Describe the bug**
When a player dodges before starting a stage animation, the dodge animation finishes AFTER the player is positioned, moving them again.

This causes the game to soft-lock in the case of level 30 and it's ladder.

**Steps to reproduce**
1. Dodge right before starting an animation
2. Enjoy

**Expected behavior**
The player should snap to the correct position where the animation is supposed to happen at.

**Screenshots or video**
https://github.com/user-attachments/assets/fe34d00e-40e0-4c8a-a88b-c07fb299ab99

**Desktop (please complete the following information):**
 - OS: Windows
 - Version 0.49.0

**Additional context**
Funnily enough, the same bug was in M:PN at launch.

**Comment by mestiez** [comment]
Duplicate of #7
